index.js

```diff
@@ -343,163 +341,159 @@ Filter.prototype._handleFile = function(relativePath, srcDir, destDir, entry, ou
 
-  instrumentation.processString++;
-  let processStringStart = process.hrtime();
-  let string = invoke(this.processor, this.processor.processString, [this, contents, relativePath, forceInvalidation, instrumentation]);
+    return string.then(outputString => {
+      instrumentation.processStringTime += nanosecondsSince(processStringStart);
+      let outputPath = filter.getDestFilePath(relativePath, entry);
 
-  return string.then(outputString => {
-    instrumentation.processStringTime += nanosecondsSince(processStringStart);
-    let outputPath = filter.getDestFilePath(relativePath, entry);
+      if (outputPath == null) {
+        throw new Error('canProcessFile("' + relativePath +
+                        '") is true, but getDestFilePath("' +
+                        relativePath + '") is null');
+      }
 
-    if (outputPath == null) {
-      throw new Error('canProcessFile("' + relativePath +
-                      '") is true, but getDestFilePath("' +
-                      relativePath + '") is null');
-    }
+      outputPath = destDir + '/' + outputPath;
 
-    outputPath = destDir + '/' + outputPath;
+      if (isChange) {
+        let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
+        if (isSame) {
+          this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
+          return;
+        } else {
+          this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+        }
+      }
 
-    if (isChange) {
-      let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
-      if (isSame) {
-        this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
-        return;
-      } else {
-        this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+      try {
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
+
+      } catch(e) {
+        // optimistically assume the DIR was patched correctly
+        mkdirp.sync(path.dirname(outputPath));
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
       }
-    }
 
-    try {
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
+      return outputString;
+    });
+  }
 
-    } catch(e) {
-      // optimistically assume the DIR was patched correctly
-      mkdirp.sync(path.dirname(outputPath));
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
-    }
+  /**
+   * @param contents {string}
+   * @param relativePath {string}
+   * @returns {string}
+   */
+  processString (contents, relativePath) { // jshint ignore:line
+    throw new Error(
+        'When subclassing broccoli-persistent-filter you must implement the ' +
```

Comment:
all errors should be prefixed with `[BroccoliPersistentFilter]` to make error identification easier.